R/manual_layer_udf.R

```diff
@@ -52,9 +52,11 @@ execute_generic_udf <- function(namespace, udf=NULL, registered_udf_name=NULL, a
   if (!is.null(args)) {
     generic_udf$argument <- jsonlite::toJSON(as.integer(serialize(args, NULL)))
   }
-  if (!is.null(result_format)) {
-    generic_udf$result_format <- ResultFormat$new(result_format)
-  }
+  stopifnot (!is.null(result_format))
```

Comment:
That's not great. `stopifnot` could/should have a clearer error, and `match.arg()` would have given you the check for free.

Reply:
Oh no! Follow-on then, sorry @eddelbuettel

Reply:
```
> x <- ResultFormat$new('lala')
Error in initialize(...) :
  Use one of the valid values: native, json, arrow
```

Reply:
re https://github.com/TileDB-Inc/TileDB-Cloud-R/pull/75#discussion_r798884765

Reply:
ok -- as long as it gets check in most if not all entry points....

I do dig 

```r
stopifnot(`please use abc def message you like here` = somePredicateBeingTrue(arg))
```

a lot though.  But that may just be me...)

Reply:
https://github.com/TileDB-Inc/TileDB-Cloud-R/pull/78